pkg/virt-handler/hotplug-disk/mount.go

```diff
@@ -346,6 +349,22 @@ func (m *volumeMounter) isDirectoryMounted(vmiStatus *v1.VirtualMachineInstanceS
 // isBlockVolume checks if the volumeDevices directory exists in the pod path, we assume there is a single volume associated with
 // each pod, we use this knowledge to determine if we have a block volume or not.
 func (m *volumeMounter) isBlockVolume(vmiStatus *v1.VirtualMachineInstanceStatus, volumeName string) bool {
+	// First evaluate the migrated volumed. In the case of a migrated volume, virt-handler needs to understand if it needs to consider the i
+	// volume mode of the source or the destination. Therefore, it evaualtes if its is running on the destiation host or otherwise, it is the source.
+	isDstHost := false
+	if vmiStatus.MigrationState != nil {
+		isDstHost = vmiStatus.MigrationState.TargetNode == m.host
+	}
+	for _, migVol := range vmiStatus.MigratedVolumes {
+		if migVol.VolumeName == volumeName {
+			if isDstHost && migVol.DestinationPVCInfo != nil && migVol.DestinationPVCInfo.VolumeMode != nil {
+				return *migVol.DestinationPVCInfo.VolumeMode != k8sv1.PersistentVolumeFilesystem
+			}
+			if !isDstHost && migVol.SourcePVCInfo != nil && migVol.SourcePVCInfo.VolumeMode != nil {
+				return *migVol.SourcePVCInfo.VolumeMode != k8sv1.PersistentVolumeFilesystem
+			}
```

Comment:
Maybe use `storagetypes.IsPVCBlock()` like below instead of `*VolumeMode != k8sv1.PersistentVolumeFilesystem`